lib/cloud/azure/kubernetes.go

```diff
@@ -0,0 +1,688 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	armazcore "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v2"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/rbac/v1"
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	authztypes "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// AKSAuthMethod defines the authentication method for AKS cluster.
+type AKSAuthMethod uint8
+
+const (
+	// AzureRBAC indicates that the Azure AD is enabled and authorization is handled by Azure RBAC.
+	AzureRBAC AKSAuthMethod = iota
+	// AzureAD indicates that the Azure AD is enabled but authorization is handled by Kubernetes RBAC.
+	AzureAD
+	// LocalAccounts indicates that the cluster access happens through Local accounts created
+	// during provisioning phase.
+	LocalAccounts
+)
+
+// AKSCluster represents an AKS cluster
+type AKSCluster struct {
```

Comment:
Nit: Is this struct needed? Why not use types.KubernetesCluster?

Disregard if it's needed.

Reply:
This structure contains Azure auth settings that are not stored in types.KubeCluster.